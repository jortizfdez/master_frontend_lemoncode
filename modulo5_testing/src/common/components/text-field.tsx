import * as React from 'react';
import MaterialTextField, {
  TextFieldProps as MaterialTextFieldProps,
} from '@material-ui/core/TextField';
import { Overwrite } from '@material-ui/types';

export interface TextFieldProps {
  name: string;
  label: string;
  value;
  onChange: (field: string, value, event?) => void;
  error?: string;
  type?: string;
  onBlur?: (field: string, value, event?) => void;
  className?: string;
}

export const TextField: React.StatelessComponent<
  Overwrite<MaterialTextFieldProps, TextFieldProps>
> = props => {
  const {
    name,
    label,
    onChange,
    value,
    error,
    type,
    onBlur,
    className,
    ...otherProps
  } = props;
  const hasError = Boolean(error);

  const handleChange = e => {
    onChange(name, e.target.value, e);
  };

  const handleBlur = e => {
    if (onBlur) {
      onBlur(name, e.target.value, e);
    }
  };

  return (
    <MaterialTextField
      {...otherProps}
      className={className}
      label={label}
      margin="normal"
      value={value}
      type={type}
      onChange={handleChange}
      onBlur={handleBlur}
      error={hasError}
      helperText={hasError && error}
    />
  );
};

TextField.defaultProps = {
  type: 'text',
};
