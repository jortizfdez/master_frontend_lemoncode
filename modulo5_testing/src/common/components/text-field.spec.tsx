import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import { TextField } from './text-field';

afterEach(cleanup);

describe('TextField component specs', () => {
    it('should display textfield with a name and label', () => {
        // Arrange
        const props = {
            name: 'Test name',
            label: 'Test label',
            value: 'Test value',
            onChange: jest.fn(),
        };

        // Act
        const { getByText } = render(<TextField {...props} />);
        const labelElement = getByText(props.label);

        // Assert
        expect(labelElement).not.toBeUndefined();
    });

    it('should display textfield using snapshot testing', () => {
        // Arrange
        const props = {
            name: 'Test name',
            label: 'Test label',
            value: 'Test value',
            onChange: jest.fn(),
        }

        // Act
        const { asFragment } = render(<TextField {...props} />);
        // Assert
        expect(asFragment()).toMatchSnapshot();
    });
});