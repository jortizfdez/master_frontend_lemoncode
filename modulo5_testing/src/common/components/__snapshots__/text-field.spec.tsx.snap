// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component specs should display textfield using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Test label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value="Test value"
      />
    </div>
  </div>
</DocumentFragment>
`;
